app/src/main/kotlin/com/vrem/wifianalyzer/wifi/scanner/Transformer.kt

```diff
@@ -63,6 +71,28 @@ internal class Transformer(private val cache: Cache) {
             listOf()
         }
 
+    internal fun fastRoaming(scanResult: ScanResult): List<FastRoaming> =
+        if (minVersionR()) {
+            scanResult.informationElements.mapNotNull {
+                if (it.id == RM_ENABLED_CAPABILITIES_IE &&
+                    validRoamingBit(it.bytes, NEIGHBOR_REPORT_IDX, NEIGHBOR_REPORT_BIT)
+                ) FastRoaming.K
+                else if (it.id == EXTENDED_CAPABILITIES_IE &&
+                    validRoamingBit(it.bytes, BSS_TRANSITION_IDX, BSS_TRANSITION_BIT)
+                ) FastRoaming.V
+                else if (it.id == MOBILE_DOMAIN_IE) FastRoaming.R
+                else null
```

Comment:
Please do not use `null` in Kotlin.
https://kotlinlang.org/docs/null-safety.html

Reply:
Here, `null` won't return, `null` is used to filter useless beacons by `mapNotNull`.

Reply:
Example:
`FastRoaming.entries.filter { it.exist(informationElement) }.toList()`

Reply:
Reconstruct and fix by latest commit.

Reply:
It is not fixed by commit 72e423d42a76c8300a670793e042982710e7b6c3

The above is an example on how to avoid nulls, there is still more code to write

Reply:
How about 1caca19a4c3db04b513628ed91db7068eb0e7a67 ? 
I'm not familiar to kotlin. Maybe you can reconstruct the implementation to a better one.